lib/pure/os.nim

```diff
@@ -1716,15 +1716,54 @@ proc createSymlink*(src, dest: string) {.noWeirdTarget.} =
     if symlink(src, dest) != 0:
       raiseOSError(osLastError(), $(src, dest))
 
+when defined(windows):
+  proc getFinalPathNameByHandleW(
+    hFile: Handle,
+    lpszFilePath: WideCStringObj,
+    cchFilePath: DWORD,
+    dwFlags: DWORD
+  ): DWORD {.importc: "GetFinalPathNameByHandleW",
+    stdcall, dynlib: "Kernel32.dll".}
+
 proc expandSymlink*(symlinkPath: string): string {.noWeirdTarget.} =
   ## Returns a string representing the path to which the symbolic link points.
-  ##
-  ## On Windows this is a noop, `symlinkPath` is simply returned.
+  ## 
+  ## .. Note:: The function fails if `symlinkPath` is not a symlink, except on Windows, 
+  ##    where the full path will be returned.
   ##
   ## See also:
   ## * `createSymlink proc <#createSymlink,string,string>`_
+  ## * `symlinkExists proc <#symlinkExists,string>`_
   when defined(windows):
-    result = symlinkPath
+    const bufsize = 32
+    const VOLUME_NAME_DOS = 0
+
+    var handle = openHandle(symlinkPath, false)
+    defer: discard closeHandle(handle)
+
+    if handle == INVALID_HANDLE_VALUE:
+      raiseOSError(osLastError(), symlinkPath)
+
+    var
+      buffer = newWideCString("", bufsize)
+      length = getFinalPathNameByHandleW(handle, buffer, bufsize, VOLUME_NAME_DOS)
+
+    buffer = newWideCString(length.int)
+    length = getFinalPathNameByHandleW(handle, buffer, length.DWORD, VOLUME_NAME_DOS)
+
+    if length == 0:
+      raiseOSError(osLastError())
+
+    result = $buffer
+
+    if length > 4:
+      if result.startsWith(r"\\?\"):
+        # In case of a local path, remove the prefix `\\?\`
+        result.setSlice(4 .. result.high)
+      elif result.startsWith(r"\\?\UNC\"):
```

Comment:
can you add a link to some docs in a comment?
(and +1 for setSlice :) )

Reply:
fixed